.github/workflows/staleness_check.yml

```diff
@@ -20,15 +25,20 @@ jobs:
       matrix:
         branch: [main, 25.x, 27.x, 28.x]
         os: [{ name: Linux, value: ubuntu-latest}]
+        exclude:
+          # If we are in a presubmit run, only test main
+          - branch: ${{ !inputs.continuous-run && '25.x' }}
+          - branch: ${{ !inputs.continuous-run && '27.x' }}
+          - branch: ${{ !inputs.continuous-run && '28.x' }}
 
-    name: Test staleness ${{ matrix.os.name }} ${{ github.head_ref && 'PR' || matrix.branch }}
+    name: Test staleness ${{ matrix.os.name }} ${{ matrix.branch }}
     runs-on: ${{ matrix.os.value }}
     if: ${{ github.event.repository.full_name == 'protocolbuffers/protobuf' }}
     steps:
-      - name: Checkout ${{ github.head_ref && 'PR' || matrix.branch }}
+      - name: Checkout ${{ matrix.branch }}
         uses: protocolbuffers/protobuf-ci/checkout@v3
         with:
-          ref: ${{ inputs.safe-checkout || github.head_ref || matrix.branch }}
```

Comment:
I think we need github.head_ref here.  Otherwise it'll test staleness on main for every PR.  The way this is setup now PRs only get `bazel query ...` run to make sure we don't submit invalid bazel files

Reply:
I don't really understand that. This variable is just for the checkout step, right? Why do we need to pass the string "Pull Request" to that instead of the branch that we want to check out?

Reply:
Ah yea I was mixing this up with the behavior below, but I think this is still wrong.    Your changes to `name:` should be fine, since you've also removed the confusing duplication (that's why I made them all say "PR").  However, we still need to check out `github.head_ref` when it's set (it only gets set for pull requests).  Otherwise we'd be testing against main instead of pending changes

Reply:
Got it, reverted this change.